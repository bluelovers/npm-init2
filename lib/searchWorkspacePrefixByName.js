"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.searchWorkspacePrefixByName = void 0;
const micromatch_1 = __importDefault(require("micromatch"));
function searchWorkspacePrefixByName({ inputName, workspacesConfig, }) {
    let workspacePrefix = workspacesConfig.prefix[0];
    if (workspacesConfig.prefix.length > 1) {
        for (let i in workspacesConfig.prefixSub) {
            const prefix = workspacesConfig.prefixSub[i];
            if (prefix.length && micromatch_1.default.isMatch(inputName, prefix + '/*')) {
                workspacePrefix = workspacesConfig.prefix[i];
                break;
            }
            if (prefix === '') {
                workspacePrefix = workspacesConfig.prefix[i];
            }
        }
    }
    return workspacePrefix;
}
exports.searchWorkspacePrefixByName = searchWorkspacePrefixByName;
exports.default = searchWorkspacePrefixByName;
//# sourceMappingURL=searchWorkspacePrefixByName.js.map