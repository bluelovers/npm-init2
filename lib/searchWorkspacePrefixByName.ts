import { parseStaticPackagesPaths } from 'workspaces-config';
import micromatch from 'micromatch';

export function searchWorkspacePrefixByName({
	inputName,
	workspacesConfig,
}: {
	inputName: string,
	cwd?: string,

	targetName?: string,
	hasWorkspace?: string,
	workspacePrefix?: string,
	workspacesConfig: ReturnType<typeof parseStaticPackagesPaths>
})
{
	let workspacePrefix: string = workspacesConfig.prefix[0];

	if (workspacesConfig.prefix.length > 1)
	{
		for (let i in workspacesConfig.prefixSub)
		{
			const prefix = workspacesConfig.prefixSub[i]

			if (prefix.length && micromatch.isMatch(inputName, prefix + '/*'))
			{
				workspacePrefix = workspacesConfig.prefix[i]
				break;
			}

			if (prefix === '')
			{
				workspacePrefix = workspacesConfig.prefix[i]
			}
		}
	}

	return workspacePrefix
}

export default searchWorkspacePrefixByName
