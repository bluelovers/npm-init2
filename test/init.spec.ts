import { parseWorkspaces, parseStaticPackagesPaths } from 'workspaces-config';
import searchWorkspacePrefixByName from '../lib/searchWorkspacePrefixByName';

const data = [
	"packages/@yarn-tool/*",
	"packages/*",
]

const workspacesConfig = parseStaticPackagesPaths(data)

it('should searchWorkspacePrefixByName', function ()
{

	let actual = searchWorkspacePrefixByName({
		inputName: '@yarn-tool/kkk',
		workspacesConfig,
	})

	expect(actual).toStrictEqual('packages/@yarn-tool')
	expect(actual).toMatchSnapshot();

	actual = searchWorkspacePrefixByName({
		inputName: 'kkk',
		workspacesConfig,
	})

	expect(actual).toStrictEqual('packages')
	expect(actual).toMatchSnapshot();

});
