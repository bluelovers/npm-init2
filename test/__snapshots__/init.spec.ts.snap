// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should searchWorkspacePrefixByName 1`] = `"packages/@yarn-tool"`;

exports[`should searchWorkspacePrefixByName 2`] = `"packages"`;
